interfaces/cython/cantera/test/test_reactor.py

```
@@ -1037,6 +1037,48 @@ class TestIdealGasConstPressureReactor(TestConstPressureReactor):
     reactorClass = ct.IdealGasConstPressureReactor
 
 
+class TestIdealGasConstPressureMoleReactor(TestIdealGasConstPressureReactor):
+    reactorClass = ct.IdealGasConstPressureMoleReactor
+
+    def test_adaptive_precon_integration(self):
+        self.create_reactors()
+        self.net2.preconditioner = ct.AdaptivePreconditioner()
+        self.net2.set_derivative_settings({"skip-third-bodies":True, "skip-falloff":True})
+        self.integrate()
```

Rather than running just a single test with the preconditioner, I think it would be better to run all of the cases from `TestConstPressureReactor` with the preconditioner. You can do that by overriding the `create_reactors` method with something like the following:
```py
    def create_reactors(self, **kwargs):
        super().create_reactors(**kwargs)
        self.net2.preconditioner = ct.AdaptivePreconditioner()
        self.net2.set_derivative_settings({"skip-third-bodies":True, "skip-falloff":True})
```
The same can be done for `TestIdealGasMoleReactor`. I tried this and saw that all the tests pass, but the test `TestIdealGasConstPressureMoleReactor::test_with_surface_reactions` is inordinately slow.

---

@speth Preconditioning is not currently implemented for surface chemistry. This is a work in progress which I intended to include in a different pull request. So, I am not quite sure why the `test_with_surface_reactions` would be majorly affected. However, I can make this change.

---

Isn't that the answer right there? You aren't providing a good estimate of the Jacobian when there are surface reactions, and therefore it barely works. I'd suggest that the implementation of `jacobian(...)` should raise an exception if there are any surfaces in the reactor. You'll then need to update the test suite to skip these specific tests where they don't apply.